libcnb-data/src/build_plan.rs

```diff
@@ -105,6 +111,35 @@ impl Require {
             metadata: Table::new(),
         }
     }
+
+    /// Convert a Serializable struct and store it as a toml Table for metadata
+    ///
+    /// # Errors
+    /// This will return error for any normal TOML serialization error as well if it's not
+    /// possible to serialize as a TOML Table.
+    pub fn metadata<T: Serialize>(&mut self, metadata: T) -> Result<(), toml::ser::Error> {
+        if let toml::Value::Table(table) = toml::Value::try_from(metadata)? {
+            self.metadata = table;
+
+            Ok(())
+        } else {
+            Err(toml::ser::Error::Custom(String::from(
+                "Could not be serialized as a TOML Table.",
+            )))
+        }
+    }
+}
+
+impl From<String> for Require {
+    fn from(s: String) -> Require {
+        Require::new(s)
+    }
+}
+
+impl From<&str> for Require {
+    fn from(s: &str) -> Require {
+        Require::new(s)
+    }
```

Comment:
This would cover both cases + more:

```suggestion
impl<S: Into<String>> From<S> for Require {
    fn from(s: S) -> Self {
        Require {
            name: s.into(),
            metadata: Default::default(),
        }
    }
}
```